packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx

```diff
@@ -337,8 +337,10 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
   }
 
   private _onDrop(item?: any, event?: DragEvent): void {
-    const draggedColumnIndex = this._draggedColumnIndex;
+    const showCheckbox = this.props.selectAllVisibility !== SelectAllVisibility.none;
+    const draggedColumnIndex = showCheckbox ? this._draggedColumnIndex : this._draggedColumnIndex + 1;

```

Comment:
I feel like this is going to be a bug farm. We should either _derive_ this._draggedColumnIndex by a function or _keep it in sync_ with the this.props.selectAllVisibility state.

Reply:
Fair -- going to try the function method in an iteration.